project-minimization-plugin/src/main/kotlin/org/plan/research/minimization/plugin/psi/PsiModificationManager.kt

```diff
@@ -0,0 +1,83 @@
+package org.plan.research.minimization.plugin.psi
+
+import com.intellij.openapi.application.EDT
+import com.intellij.openapi.command.writeCommandAction
+import com.intellij.openapi.components.Service
+import com.intellij.openapi.project.Project
+import mu.KotlinLogging
+import org.jetbrains.kotlin.psi.*
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+/**
+ * Service that provides functions to modify the bodies of various Kotlin elements within a project.
+ *
+ */
+@Service(Service.Level.PROJECT)
+class PsiModificationManager(private val rootProject: Project, private val cs: CoroutineScope) {
```

Comment:
It's needed to make a good way to wait for transformations end

Reply:
IMO using EDT is a good way to do it

Reply:
It really depends on how coroutines and EDT schedulers work, and basically, if you execute several coroutines on the same thread, there is no guarantee that these jobs will wait for each other

Reply:
AFAIK for `invokeLater` it is guaranteed to form a queue, I'm not sure about coroutines, I'll add some synchronization